R/rows.R

```diff
@@ -112,16 +112,20 @@ rows_update.data.frame <- function(x, y, by = NULL, ..., copy = FALSE, in_place
   y <- auto_copy(x, y, copy = copy)
   rows_df_in_place(in_place)
 
-  rows_check_key_df(x, key, df_name = "x")
+  rows_check_key_df(x, key, df_name = "x", .check_unique = FALSE)
   rows_check_key_df(y, key, df_name = "y")
-  idx <- vctrs::vec_match(y[key], x[key])
 
-  bad <- which(is.na(idx))
-  if (has_length(bad)) {
+  if (!all(vec_in(y[key], x[key]))){
```

Comment:
Do we even need this check?

```suggestion
  if (FALSE) {
```

If we introduce an asymmetry here between zero and nonzero matches, we contradict recycling rules in vctrs.

The current implementations requires one and only one match. No recycling. It's safe and limiting.

The proposed implementation recycles one match to the number of entries in the target table, but fails if the target table has zero entries.

- Avoiding the check entirely makes us more consistent with recycling rules
- One use case of `rows_update()` might be imputation: we update `NA` or otherwise bad values in existing columns in `x` based on a lookup table in `y`. Why should that fail when some of the looked-up values are missing in `x`?
- The SQL version `UPDATE target SET ... FROM x, y WHERE x.key == y.key` also silently ignores missing entries in `x`